### PR TITLE
Add mypy support and integration to builds

### PR DESCRIPTION
**Pull request by TimOrme**
Adds mypy integration to CI/CD and pre-commit.  Fixes any issues found there.

### CODE REVIEW

File: philter_lite/coordinate_map.py [hunk 1]
```diff
@@ -225,8 +226,8 @@ def to_ranges(iterable):
             for key, group in itertools.groupby(
                 enumerate(iterable), lambda t: t[1] - t[0]
             ):
-                group = list(group)
-                yield group[0][1], group[-1][1] + 1
+                group_list = list(group)
```
Comment (skucherlapati) [inline philter_lite/coordinate_map.py:229]: can group_list be None?

**Review by skucherlapati — APPROVED** [review]
LGTM - except a minor question. Not sure if `to_ranges` needs null check. Looks like a function inside a function so may be it is not accessible from outside so it is meant to be used with only a certain object type?

File: philter_lite/philter.py [hunk 2]
```diff
@@ -380,10 +380,7 @@ def _get_exclude_include_maps(
     data_tracker: DataTracker,
 ):
     exclude = pattern.exclude
-    if hasattr(pattern, "filepath"):
-        filter_path = pattern.filepath
-    else:
-        filter_path = pattern.title
+    filter_path = pattern.title
```
Comment (skucherlapati) [inline philter_lite/philter.py:383]: currently pattern title is just a string - are we going to change it moving forward after this gets into master?

File: philter_lite/coordinate_map.py (hunk 1 above)
Comment (TimOrme) [reply philter_lite/coordinate_map.py:229]: I dont think so, think itertools is always going to provide at a minimum an empty list, and even that I can't think of a case where it wouldn't have some values in it.

File: philter_lite/philter.py (hunk 2 above)
Comment (TimOrme) [reply philter_lite/philter.py:383]: Yeah we no longer reference things by filepath any more, just by title.  There's still some old naming in here, but at least the value in the PHI entries is single use now.  I'll try and clean up some of those other pieces later on, for the things below like the PHI results.  I don't think the filepath attribute on those is even used currently.

**Comment by TimOrme** [comment]
> LGTM - except a minor question. Not sure if `to_ranges` needs null check. Looks like a function inside a function so may be it is not accessible from outside so it is meant to be used with only a certain object type?
> 
> (edit) - Found that pattern.title is assigned to filter_path. Are we going to change the configs moving forward? title is not very intuitive for file path name - or perhaps I misunderstood....

`to_ranges` should only be called internally there and the typing on it should always be a list of some kind.  A lot of these need some restructuring to properly type check and this is probably one of those cases, but trying not to change too much in one go.

**Pull request merged**